### PR TITLE
[problem4] initial commit.

### PR DESCRIPTION
**Pull request by daveholtz**
@mfzhao 

I think this code should work, but it doesn't seem to be working. Want to take a look and see if you spot any obvious issues? Looks like the thetas just get bigger and bigger...

### CODE REVIEW

**Comment by daveholtz** [comment]
@mfzhao I think this code does most of what needs to be done to write up this problem - would be good to double check the loss functions, as the MSE is lower than I would expect on the training data (especially based on the graph in the p-set).
